### ISSUE TITLE
fiminus should not be defined as (fiplus _ (fineg _))

### ISSUE THREAD

**Issue by andrew-appel** [validsdp/validsdp#23]
In `libvalidsdp/float_infnan_spec.v` there is the line,
```
Definition fiminus (x y : FIS fs) : FIS fs := fiplus x (fiopp y).
```
Unfortunately, in IEEE floating point, (x-y) is not always the same as (x + (-y)); it may raise a different NaN.  They are the same in the BinarySIngleNaN model, but not in the general Flocq Binary model.  If we want libvalidsdp (its models and its proofs) to be generally useful for reasoning about C programs, then either

- we rewrite all the C programs to use (x + (-y)) instead of (x-y) ; or 
- we add a subtraction operator to `Record Float_infnan_spec`
- There is a third alternative, which is rather painful:  in the midst of the proofs that the C programs correctly and exactly implement the floating-point functional models, prove the absence of NaNs.  But sometimes there may be NaNs, and even when not, it's painful to carry around the invariants that would prove their absence.

**Comment by proux01** [comment]
Indeed, we should probably add some fiminus with a spec similar to addition plus a spec that on finite values x - y matches x + (- y). This whole NaN business is a real pain but I guess we cannot escape it considering the way Compcert is made.

**Comment by andrew-appel** [comment]
Don't blame CompCert.  CompCert is just matching the semantics of the computer architectures:  x86, ARM, RISC-V, etc.
